### ISSUE TITLE
Decouple ReplicationTransfer and BagManRequest models

### ISSUE THREAD

**Issue by dazza-codes** [dpn-admin/dpn-server#80]
Extracted from https://github.com/dpn-admin/dpn-server/pull/77#pullrequestreview-1620865

The control flow, transactionality or atomicity, of the `cancel!` operations between the `ReplicationTransfer` and the `BagManRequest` are complex because they each call the other and, without the `unless cancelled` statement, might incur an infinite loop. I wonder whether the calls from one model to the other should be extracted out of the models, to allow an atomic update of one model and then another model? Perhaps this coupling between these models should be extracted from the model level up to the controller level?  An additional decoupling would be to have a client manage all aspects of the `BagManRequest` update, to provide or manage eventual consistency with the DPN REST API updates for `ReplicationTransfer`.

**Comment by darrenleeweber** [comment]
Some of the specs could be masking the coupling between the models.  For example, the `spec/models/bag_man_request_spec.rb ` has some mocks that replace the original calls, like
```ruby
expect(replication).to receive(:cancel!)
```
This effectively makes `replication.cancel!` a non-op.  If it were changed so that it calls the original method, we might see rspec failures related to the number of times the `cancel!` methods are called in the models. So, if it were:
```ruby
expect(replication).to receive(:cancel!).and_call_original
```
then the `Replication#cancel!` method would try to call (back to) the `BagManRequest#cancel!` method.  That should amount to 2 calls, one in the spec and another loop-back call.  The 2nd loop-back call should not execute any db changes, so long as `BagManRequest#cancelled` returns `true`.

**Comment by dazza-codes** [comment]
This can be addressed after #83

**Comment by malakai97** [comment]
Hathi is now using an implementation that completely eliminates BagManRequest.  I recommend using that going forward.  Putting this on hold.

**Comment by Pcolar** [comment]
Does your change alter the replication state diagram?  Assuming it was correct to begin with...
I would like to work with you to complete this documentation.
https://www.lucidchart.com/invitations/accept/60d79689-313d-4464-96fc-dfe4d8d638d1

**Comment by dazza-codes** [comment]
There is also decoupling of the REST API and the bag transfers by using the https://github.com/dpn-admin/dpn-sync project.
